src/Data/Unrestricted/Internal/Instances.hs

```diff
@@ -351,3 +361,74 @@ instance (Movable a, Prelude.Semigroup a) => Semigroup (MovableMonoid a) where
       combine (Ur x) (Ur y) = x Prelude.<> y
 
 instance (Movable a, Prelude.Monoid a) => Monoid (MovableMonoid a)
+
+instance Consumable (RepStream a) where
+  consume (RepStream givex dupsx consumesx sx) = consumesx sx
+instance Dupable (RepStream a) where
+  dupR (RepStream givex dupsx consumesx sx) = CollectFrom $ RepStream
+    (\sx' -> RepStream givex dupsx consumesx sx')
+    dupsx
+    consumesx
+    sx
+
+instance Consumable (Replicator a) where
+  consume = \case
+    Moved _ -> ()
+    CollectFrom stream -> consume stream
+
+instance Dupable (Replicator a) where
+  dupR = \case
+    Moved x -> Moved (Moved x)
+    CollectFrom stream -> CollectFrom Data.<$> dupR stream
+
+instance Data.Functor RepStream where
+  fmap f (RepStream givef dupsf consumesf sf) =
+    RepStream (f . givef) dupsf consumesf sf
+instance Data.Applicative RepStream where
+  pure x = RepStream
+    unur
+    (\case
+      Ur x -> (Ur x, Ur x))
+    (\case
+      Ur _ -> ())
+    (Ur x)
+  (RepStream givef dupsf consumesf sf) <*> (RepStream givex dupsx consumesx sx) =
+    RepStream
+      (\(sf', sx') -> givef sf' (givex sx'))
+      (\(sf', sx') ->
+        (dupsf sf', dupsx sx') & \case
+          ((sf1, sf2), (sx1, sx2)) -> ((sf1, sx1), (sf2, sx2))
+      )
+      (\(sf', sx') -> consumesf sf' & \case
+        () -> consumesx sx')
+      (sf, sx)
+
+instance Data.Functor Replicator where
+  fmap f = \case
+    Moved x -> Moved (f x)
+    CollectFrom stream -> CollectFrom $ Data.fmap f stream
+instance Data.Applicative Replicator where
+  pure = Moved
+  (Moved f) <*> (Moved x) = Moved (f x)
+  sf <*> sx = CollectFrom (toRepStream sf Data.<*> toRepStream sx)
+
+toRepStream :: Replicator a %1 -> RepStream a
+toRepStream = \case
+  Moved x -> Data.pure x
+  CollectFrom stream -> stream
+
+nextR :: Replicator a %1 -> (# a, Replicator a #)
+nextR (Moved x) = (# x, Moved x #)
+nextR (CollectFrom (RepStream givex dupsx consumesx sx)) =
+  dupsx sx & \case
+    (sx1, sx2) -> (# givex sx1, CollectFrom (RepStream givex dupsx consumesx sx2) #)
+
+elim' :: forall n a b. KnownNat n => Replicator a %1 -> FunN n a b %1 -> b
```

Comment:
Ah, hence the export before. This points to a need or reordering things a little.